src/modules/battery_status/battery_status.cpp

```diff
@@ -171,11 +170,7 @@ BatteryStatus::parameter_update_poll(bool forced)
 void
 BatteryStatus::adc_poll()
 {
-	/* make space for a maximum of twelve channels (to ensure reading all channels at once) */
-	px4_adc_msg_t buf_adc[PX4_MAX_ADC_CHANNELS];
-
-	/* read all channels available */
-	int ret = px4_read(_adc_fd, &buf_adc, sizeof(buf_adc));
+	adc_report_s adc_report = {};
```

Comment:
```suggestion
	adc_report_s adc_report;
```
And please move this down before `if (_adc_report_sub.update(&adc_report)) {`